### PR TITLE
Responsiveness/merged-with-update-navi

### PR DESCRIPTION
**Pull request by prince-hope1975**
- Modified viteconfig for local development and path aliases
- Added tailwind style utility classes to to custom.scss file
- Added responsive design to all pages, and included a mobile navbar
- Removed overflow from long text and enabled copying text

## Release Notes

- **New Features**
  - Introduced `CopyText` component for easy text copying across various sections.
  - Added `MobileMenu` for improved mobile navigation.
  - New responsive design adjustments for enhanced layout and usability.
  - Added new utility classes for styling flexibility in `custom.scss`.
  - Introduced a new `Table` component in the Airdrop section to display relevant data.

- **Enhancements**
  - Updated styling for various components, including the overview and stake sections, to improve responsiveness and visual clarity.
  - Enhanced `WalletWidget` to display a shortened account address based on screen size.
  - Improved layout and state management in the `Withdraw` component for better user experience.

- **Bug Fixes**
  - Addressed layout issues in the `Withdraw` component for better user experience.

- **Chores**
  - Updated project dependencies for improved performance and security.
  - Enhanced TypeScript configuration for better development experience.

### CODE REVIEW

File: apps/web/src/Components/Copy/index.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  return (
+    <Wrapper className="relative">
+       
+      <Copy onClick={handleCopy} size={15} />
+      {copied&&<div className="absolute copied">Copied</div>}
+    </Wrapper>
+  );
+};
+
+export default CopyText;
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Components/Copy/index.tsx:42]: This is helpful thanks

File: apps/web/src/Components/LeftPanel/LeftPanel.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
+import React from "react";
 import "./LeftPanel.scss";
 import { Tab, Tabs } from "@mui/material";
 import { Link, useLocation } from "react-router-dom";
-import React, { ReactElement, useEffect } from "react";
+import { ReactElement, useEffect } from "react";
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Components/LeftPanel/LeftPanel.tsx:5]: The global import is used by text editors like vscode, and would display some locally when not present

File: apps/web/src/Components/LeftPanel/LeftPanel.tsx [hunk 3]
```diff
@@ -16,6 +16,11 @@ import { useWallet } from "@txnlab/use-wallet-react";
 import { loadAccountData } from "../../Redux/staking/userReducer";
 import { useAppDispatch } from "../../Redux/store";
 import LockClockIcon from "@mui/icons-material/LockClock";
+import {
+  dashboardRoutes,
+  dashboardTabs,
+  settingRoutes,
+} from "../../constants/routes";
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Components/LeftPanel/LeftPanel.tsx:23]: The import is being used in the specific file on line 92

File: apps/web/src/Components/MobilePanel/index.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  const location = useLocation();
+
+  const theme = "dark";
+  const navlinks: string[] = [
+    ...dashboardRoutes,
+    ...airdropRoutes,
+    ...stakingRoutes,
+    ...settingRoutes,
+  ];
+  const [open, setOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Components/MobilePanel/index.tsx:29]: Ok

File: apps/web/src/Pages/Overview/Overview.scss [hunk 5]
```diff
@@ -1,63 +1,115 @@
...
 
-.overview-wrapper {
+.overview-wrapper-component {
+  $sm: 640px;
+  $md: 768px;
+  $lg: 1024px;
+  $xl: 1280px;
+  $xxl: 1536px;
+  $side-width: 250px;
+  $padding: 0.5rem;
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Pages/Overview/Overview.scss:10]: The way the project is structured, it doesn't work to import it globally
If you have a way to do it please let me know.
It currently throws an error in development

File: apps/web/src/Pages/Overview/Overview.scss [hunk 6]
```diff
@@ -1,63 +1,115 @@
...
-          display: flex;
-          justify-content: flex-start;
-          align-items: center;
+
           column-gap: 20px;
           .key {
-            font-size: 14px;
-            width: 150px;
+            font-size: 1rem;
+            // width: 150px;
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Pages/Overview/Overview.scss:71]: Ok

File: apps/web/src/Pages/Airdrop/Table/Table.tsx [hunk 7]
```diff
@@ -101,14 +103,27 @@ const CompoundInterest: React.FC<CompoundInterestProps> = (
     props.time,
     props.compoundingsPerYear
   );
+  const increase = A - props.principal;
+  const percentIncrease = Math.round(
+    calculatePercentIncrease(props.principal, A) + 1
+  );
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Pages/Airdrop/Table/Table.tsx:109]: @temptemp3 
Please answer the question, as i believe this was your implementation

File: apps/web/src/Pages/Airdrop/Table/Table.tsx [hunk 8]
```diff
@@ -162,268 +177,302 @@ const AirdropTable: React.FC<LockupProps> = ({
...
+                      Number(contracts[0]?.global_period_seconds) *
                       1000,
                     { units: ["y"], round: true }
                   )}
-                </TableCell>
+                </Box>
+              </Stack>
+            </Fade>
+          )
+        ) : null}
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Pages/Airdrop/Table/Table.tsx:300]: This has a healthy amount of tenary operators

File: packages/styles/styles/custom.scss [hunk 9]
```diff
@@ -31,4 +32,732 @@
...
+}
+.py-2 {
+  padding-block: 0.5rem;
+}
+.py-2\! {
+  padding-block: 0.5rem !important;
+}
+.\!hidden {
+  display: none !important;
+}
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply packages/styles/styles/custom.scss:149]: The escape characters "\" allows for it's use
In this case it's being used to signify importance of a utility class

File: apps/web/src/constants/routes.tsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  "deposit",
+  "withdraw",
+  "transfer",
+  "delegate",
+];
+export   const airdropRoutes: string[] = ["airdrop"];
+
+export   const stakingRoutes: string[] = ["staking"];
+
+export   const settingRoutes: string[] = ["setting"];
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/constants/routes.tsx:18]: The seperate routes may get expanded in the future, so we seperate them now for the future

File: apps/web/src/Pages/Overview/Withdraw/Withdraw.tsx [hunk 11]
```diff
@@ -353,7 +353,7 @@ function Lockup({ show, onClose }: LockupProps): ReactElement {
                               color={"primary"}
                               size={"large"}
                               onClick={() => {
-                                Withdraw(accountData);
+                                deposit(accountData);
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Pages/Overview/Withdraw/Withdraw.tsx:356]: @temptemp3 Take a  look at this

File: apps/web/src/Pages/Overview/Withdraw/Withdraw.tsx [hunk 12]
```diff
@@ -93,14 +93,15 @@ function Lockup({ show, onClose }: LockupProps): ReactElement {
...
         20,
         voiStakingUtils.network.getAlgodClient()
       );
 
-      setTxnId(transaction.txID());
-      setTxnMsg("You have withdrawn successfully.");
-      dispatch(loadAccountData(activeAccount.address));
+      setTxnId(txnId);
+      setTxnMsg("You have deposited successfully.");
+      //dispatch(loadAccountData(activeAccount.address));
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply apps/web/src/Pages/Overview/Withdraw/Withdraw.tsx]: There's no commente out code here
But thank you for the suggestion

File: packages/styles/styles/custom.scss [hunk 13]
```diff
@@ -31,4 +32,722 @@
...
+
+@media (min-width: 1024px) {
+  .lg\:flex-1 { flex: 1 1 0%; }
+  .lg\:gap-4 { gap: 1rem; }
+  .lg\:truncate {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
```
[the remark this answers is not in the document]
Comment (prince-hope1975) [reply packages/styles/styles/custom.scss:751]: We can revisit this at a later date, but for now, this will do

**Review by temptemp3 — APPROVED** [review]

**Pull request merged**